### PR TITLE
Bugfix: make sure 0 byte sized binary resources can be included in ZiPfile

### PR DESCRIPTION
**Pull request by dizzzz**
Bugfix: make sure 0 byte sized binary resources can be included in ZIP file. Fixes 3565

### CODE REVIEW

File: extensions/modules/compression/src/main/java/org/exist/xquery/modules/compression/AbstractCompressFunction.java [hunk 1]
```diff
@@ -394,23 +394,23 @@ private void compressResource(OutputStream os, DocumentImpl doc, boolean useHier
...
-			// binary file
-            try (final InputStream is = context.getBroker().getBinaryResource((BinaryDocument)doc);
-                 final UnsynchronizedByteArrayOutputStream baos = new UnsynchronizedByteArrayOutputStream(doc.getContentLength() == -1 ? 1024 : (int)doc.getContentLength())) {
+            value = strDoc.getBytes();
+        } else if (doc.getResourceType() == DocumentImpl.BINARY_FILE && doc.getContentLength() > 0) {
+            // binary file
+            try (final InputStream is = context.getBroker().getBinaryResource((BinaryDocument) doc);
+                 final UnsynchronizedByteArrayOutputStream baos = new UnsynchronizedByteArrayOutputStream(doc.getContentLength() == -1 ? 1024 : (int) doc.getContentLength())) {
                 baos.write(is);
                 value = baos.toByteArray();
```
Comment (adamretter) [inline extensions/modules/compression/src/main/java/org/exist/xquery/modules/compression/AbstractCompressFunction.java:410]: What does this produce when `doc.getContentLength() == 0`? I would have thought `byte[0]`

Comment (dizzzz) [reply extensions/modules/compression/src/main/java/org/exist/xquery/modules/compression/AbstractCompressFunction.java:410]: I did not check, but the `write()` never ends

Comment (dizzzz) [reply extensions/modules/compression/src/main/java/org/exist/xquery/modules/compression/AbstractCompressFunction.java:410]: `doc.getContentLength() == -1` is never true btw

Comment (dizzzz) [reply extensions/modules/compression/src/main/java/org/exist/xquery/modules/compression/AbstractCompressFunction.java:410]: at least now `byte[0]` is created :-)

**Review by adamretter — APPROVED** [review]

**Pull request merged**